File: frontend/src/api/models/Post.ts
class Post {
    id: string
    title: string
    summary: string
    text: string
    date: string
  
    constructor(data: Post) {
      this.id = data.id
      this.title = data.title
      this.summary = data.summary
      this.text = data.text
      this.date = data.date
    }
  }
  
  export default Post
  

File: backend/src/main/java/ee/fakeplastictrees/blog/post/model/Post.java
package ee.fakeplastictrees.blog.post.model;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Set;

@Data
@Builder
@Document("posts")
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Post {
    @Id
    String id;

    String title;

    String summary;

    String text;

    String date;

    Boolean visible;

    Set<String> category;
}

